lib/puppet/provider/parsedfile.rb

```diff
@@ -465,13 +456,7 @@ def initialize(record)
     # The 'record' could be a resource or a record, depending on how the provider
     # is initialized.  If we got an empty property hash (probably because the resource
     # is just being initialized), then we want to set up some defaults.
-    @property_hash = self.class.record?(resource[:name]) || {:record_type => self.class.name, :ensure => :absent} if @property_hash.empty?
```

Comment:
In my testing it's not needed to set the properties when a `Puppet::Resource` is passed in, but it is needed to set the `record_type` to skip things. Perhaps `skip_record?` isn't needed at all anymore and can be dropped. Then this line can also be dropped.